posts/2020-08-16-ocp-log-forwarding.adoc

```diff
@@ -0,0 +1,390 @@
+  port 8088
+  token 00000000-0000-0000-0000-0000000000000
+
+  # metadata parameter
+  default_source openshift
+
+  use_ssl true
+  ca_file /path/to/ca.pem
+</match>
+```
++
+*source* directive determines the input sources. It uses *forward* type to accept TCP packets from your OpenShift Container Platform. *match* directive determines the output destinations. It looks for events with matching tags and uses *splunk_hec* to sends the events to Splunk using HTTP Event Collector. Splunk's *host* and *port* are required. *token* should be replaced by Splunk's generated token. Set *use_ssl* to true to use SSL when connecting to Splunk.
+
+. Create `secure-forward.conf` to use the Fluentd forward protocol.
+* Sample `secure-forward.conf`:
++
+```
+<store>
+  @type forward
+  <security>
+    self_hostname ${hostname}
+  </security>
+
+  transport tls
+  tls_verify_hostname false
+
+  tls_cert_path '/etc/ocp-forward/ca-bundle.crt'
+
+  <server>
+    host splunk-fluentd-forward.offcluster.com
+    port 24224
+  </server>
+</store>
+```
++
+*store* plugin forwards the logs to specified outputs under *server* directive. Specify *transport* tls if you wish to enable TLS validation and specify the path to private CA certificate file using *tls_cert_path*.
+
+. Login to your OpenShift through command line tool. Create a ConfigMap named *secure-forward* in the *openshift-logging* namespace from the configuration file:
++
+[source]
+----
+[root@ocp ~]# oc create configmap secure-forward --from-file=secure-forward.conf -n openshift-logging
+----
++
+If you need to import secrets required for the receiver, also run the following command:
++
+[source]
+----
+[root@ocp ~]# oc create secret generic secure-forward --from-file=<certFile>=cert_file_from_fluentd_receiver
+----
++
+
+. Refresh the *fluentd* Pods to apply the secure-forward secret and secure-forward ConfigMap:
++
+[source]
+----
+oc delete pod --selector logging-infra=fluentd
+----
++
+
+=== Setting up Splunk Dashboard
+
+. Go to Search & Reporting. Search for `index="openshift"` for *Option 1* and `source="openshift"` for *Option 2* to view logs from OpenShift Container Platform.
+
+. Download Splunk dashboards for Open Liberty: link:https://github.com/WASdev/sample.dashboards/tree/master/Liberty/Splunk%208[Sample dashboard for Liberty using Splunk]. Import downloaded sample dashboards using *Source* option. Using this dashboard, you can visualize message, trace, and first failure data capture (FFDC) logging data collected from JSON logging in Open Liberty.
+
+image::/img/blog/splunk-dashboard.png[Splunk-Dashboard,width=70%,align="center"]
+
+== External Elasticsearch and Kibana configuration
+Using `elasticsearch` output, you can forward logs to Elasticsearch cluster deployed on your external machine. You run Elasticsearch and Kibana with docker compose. This way, obtaining docker images, setting up configurations and bringing up the clusters can be done  using one YAML file.
```

Comment:
mention that the docker compose approach to setting up elasticsearch and kibana is just for demo purposes